src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -14,7 +14,7 @@ namespace NuGet.PackageManagement.VisualStudio
 {
     [Export(typeof(ISettings))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    public class VSSettings : ISettings
+    public class VSSettings : ISettings, IDisposable
```

Comment:
Read [this](https://docs.microsoft.com/en-us/dotnet/api/system.idisposable?view=netframework-4.8#idisposable-and-the-inheritance-hierarchy).

Ideally, `VSSettings` would be `sealed`, but if it can't be, provide a `protected virtual void Dispose(bool)` method and make sure to call it in `Dispose()`.

Reply:
serves me right for copying how some of our other classes implement IDisposable. I'll do it properly and create a task to do the others later.